src/utilities/createRouter.ts

```diff
@@ -154,8 +153,23 @@ export function createRouter<const T extends Routes>(routes: T, options: RouterO
     app.provide(routeHookStoreKey, hooks)
   }
 
+  const route2 = new Proxy(route, {
+    get(target, prop, receiver) {
+      if (prop === 'params') {
+        return new Proxy(route.params, {
+          set(target, prop, value) {
+            router.push(route.key, { ...target, [prop]: value })
+            return true
+          },
+        })
+      }
+
+      return Reflect.get(target, prop, receiver)
+    },
+  })
```

Comment:
WIP

Notes: 
- Setting a param calls `router.push` and does not update the current route directly. This means the url is the source of truth. But it could be problematic if the route has async before hooks. Because the page won't react until after the route has been updated. But that would only happen in an onBeforeRouteUpdated hook. Which seems rare. 
- I'm using `router.push` rather than just `push` because otherwise we cannot spy on router.push in unit tests. 
- I'm thinking we might introduce a router setting for the push functionality here. I feel like everyone has different opinions on whether this should be a push or a replace. And we can easily allow users to customize this behavior.

Reply:
Thought: This probably breaks if you update multiple params at the same time. Line 161 will overwrite the previous updates...